src/main/java/uk/ac/ebi/ega/egafuse/runner/CommandLineOptionParser.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ega.egafuse.runner;
+
+import joptsimple.OptionParser;
+
+public class CommandLineOptionParser {
+    public static final String OPTIONS_HELP = "h";
+    
+    private static final OptionParser optionParser = buildParser();
+
+    private static OptionParser buildParser() {
```

Comment:
As I've commented in other parts. I think this class either should be an object with full state or you should only have a statuc puclic method to "parse" that returns a typed object with all the appropriated values already parsed and controlled.

Reply:
Converted the class to return the input object and handled validations here